app/channel_test.go

```diff
@@ -1901,6 +1904,34 @@ func TestPatchChannelModerationsForChannel(t *testing.T) {
 		assert.Contains(t, higherScopedGuestRole.Permissions, createPosts)
 	})
 
+	t.Run("Updates the authorization to create post", func(t *testing.T) {
+		addCreatePosts := []*model.ChannelModerationPatch{
+			{
+				Name: &createPosts,
+				Roles: &model.ChannelModeratedRolesPatch{
+					Members: model.NewBool(true),
+				},
+			},
+		}
+		removeCreatePosts := []*model.ChannelModerationPatch{
+			{
+				Name: &createPosts,
+				Roles: &model.ChannelModeratedRolesPatch{
+					Members: model.NewBool(false),
+				},
+			},
+		}
+
+		mockSession := model.Session{UserId: user.Id}
+
+		_, err := th.App.PatchChannelModerationsForChannel(channel.DeepCopy(), addCreatePosts)
+		require.Nil(t, err)
+		require.True(t, th.App.SessionHasPermissionToChannel(mockSession, channel.Id, model.PERMISSION_CREATE_POST))
+
+		_, err = th.App.PatchChannelModerationsForChannel(channel.DeepCopy(), removeCreatePosts)
+		require.Nil(t, err)
```

Comment:
We now tend to use `NoError` instead of `Nil`.

Reply:
`(*App).PatchChannelModerationsForChannel` returns a `*model.AppError`, and I think it's like this:

* `error` interfaces should `require.Error` or `require.NoError`, and
* `model.AppError` interfaces should `require.Nil` or `require.NotNil`

Unless I'm wrong?

Reply:
Oh interesting. I was convinced `Error/NoError` did actual checks on the interface but instead they merely check for nil equality. Looks good then :+1: